sdk/azcore/arm/runtime/policy_trace_namespace.go

```diff
@@ -0,0 +1,31 @@
+//go:build go1.18
+// +build go1.18
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package runtime
+
+import (
+	"net/http"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/arm/internal/resource"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/internal/shared"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/tracing"
+)
+
+// httpTraceNamespacePolicy is a policy that adds the az.namespace attribute to the current Span
+func httpTraceNamespacePolicy(req *policy.Request) (resp *http.Response, err error) {
```

Comment:
For data-plane we do that in the `azcore.NewClient()` constructor.  See the `TracingNamespace` field in the `PipelineOptions` parameter.  This would be set by SDK developers.

For ARM (the highlighted change) there is no `PipelineOptions` client constructor parameter.  Plus ARM is 100% codegen.